### PR TITLE
Dockerize the app

### LINKED ISSUE #3: Dockerize app
Local installs have been a bear, so let's have something that gets people going with a docker command.

### PR DESCRIPTION
**Pull request by LindsayYoung**
We want a much simpler local set up, so this should be pretty minimal. 

This is dependent on merging #6  first, since I wanted to make sure this would work with that set up. So the work is really the last 3 commits.

I feel confident on the python setup but I don't think the NPM set up is particularly elegant and I am open to feedback. 

this will close #3

### CODE REVIEW

**Comment by alexsoble** [comment]
I'll start reviewing here! 

I'll follow the local setup steps with Docker, see how things look for me locally, and add notes or comments if I notice anything missing from the local setup instructions or any other major issues.

**Comment by alexsoble** [comment]
I get the following error on the `docker-compose up -d --build` step:

```
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7fba6cd2c8d0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pipenv/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7fba6cc55cd0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pipenv/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7fba6cc55a90>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pipenv/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7fba6c4646d0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pipenv/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7fba6c464790>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pipenv/
ERROR: Could not find a version that satisfies the requirement pipenv (from versions: none)
ERROR: No matching distribution found for pipenv
ERROR: Service 'web' failed to build: The command '/bin/sh -c pip install pipenv' returned a non-zero code: 1
```

Because the error messages mentioned networking issues, I tried running this command both on and off the GSA network. I got the same error message both times. Will check and see if this is some kind of issue on pipenv's side and then will look at possible issues on my side. @LindsayYoung, did you see anything like this during your own setup?

**Comment by alexsoble** [comment]
Trying a hard refresh of my Docker settings.

**Comment by alexsoble** [comment]
The hard refresh seems promising... the Docker script got past `RUN pipenv install --system`...

File: README.md [hunk 1]
```diff
@@ -1,94 +1,79 @@
...
+To build the project
 
-    CREATE USER username WITH PASSWORD 'userpass' CREATEDB;
-    GRANT permissions ON DATABASE crt_portal TO username;
+    docker-compose up -d --build
 
+To run the project
 
-(I would like the local set up to be in docker next. so we don't need to to the previous steps)
+    docker-compose u
```
Comment (alexsoble) [inline README.md]: @LindsayYoung Should this be `docker-compose up`?

**Comment by alexsoble** [comment]
@LindsayYoung The Docker build and up steps are working successfully for me now! 😄

Not necessarily a blocker, but when I visit `http://0.0.0.0:8000/` I get a Django URL error. I found a real USWDS-styled response page when I visited `http://0.0.0.0:8000/form/`. We could add a note about visiting  `http://0.0.0.0:8000/form/` to the README.

File: README.md [hunk 2]
```diff
@@ -1,94 +1,79 @@
...
 
-    python manage.py migrate
 
-create a superuser for admin access
 
-    python manage.py createsuperuser
+## cloud.gov set up
 
-run the server on localhost:8000
+- create postgres DB and S3 with development settings:
```
Comment (alexsoble) [inline README.md]: @LindsayYoung Is the function/purpose of the S3 bucket to hold static assets like JS and CSS files? Could we add a note about that here?

File: README.md [hunk 3]
```diff
@@ -1,94 +1,79 @@
...
+Install Docker
 
+    https://www.docker.com/get-started
 
-In pgcli or your sql shell command to create datebase with:
 
-    create database crt_portal;
+Use a long, random string for your secret key and save it in your local environment.
 
+    export SECRET_KEY=''
```
Comment (alexsoble) [inline README.md]: A nice little developer experience improvement would be to generate and set a random SECRET_KEY automatically during local development — @LindsayYoung give this comment a 👍react if you want me to make an issue for that.

File: README.md (hunk 1 above)
Comment (LindsayYoung) [reply README.md]: yep!

File: README.md (hunk 3 above)
Comment (LindsayYoung) [reply README.md]: I thought about it but thought it is some thing we can add later. Feel free to make a ticket!

**Comment by alexsoble** [comment]
This is looking good to me — I can follow the instructions to get the site up and running with Docker, and see the correct assets for both USWDS (`/form`) and Django admin (`/admin`).

One detail I was wondering about as I looked at the files — are we using the `Dockerfile` in this branch? The Dockerfile includes the line `WORKDIR /code`, but we don't have a /code directory.

**Comment by alexsoble** [comment]
@LindsayYoung Opened up two small PRs into this PR based on what I noticed in my review — feel free to merge in one, both, or neither. #14 and #13.

File: README.md (hunk 3 above)
Comment (alexsoble) [reply README.md]: Addressed in #13, it's just one line of code so I figured a PR was easier than a ticket.

Comment (LindsayYoung) [reply README.md]: I just pushed a change to generate a random default to make the setting optional.

Comment (alexsoble) [reply README.md]: OK!

**Comment by LindsayYoung** [comment]
Great observation about the working directory! 

The working directory is where the code gets copied over it in the container.
If the project is running, in a new terminal run 

    docker exec -it crt-django_web_1 /bin/bash

That will ssh you into the container. If you then see what directory you are in with `pwd`, you will see you are in `/code`

File: README.md (hunk 2 above)
Comment (LindsayYoung) [reply README.md]: We talked abut this and documenting that we should add better explanations once we have cloud.gov running static assets in #9

**Comment by alexsoble** [comment]
@LindsayYoung Ah, I thought WORKDIR was on the local machine, not the Docker image. Thank you for explaining!

**Comment by alexsoble** [comment]
@LindsayYoung Let's merge!

**Pull request merged**